src/ThriftSQL/Hive.php

```diff
@@ -31,6 +31,15 @@ public function connect() {
     if ( null !== $this->_sessionHandle ) {
       return $this;
     }
+
+    // Make sure we have auth info set
+    if ( empty( $this->_username ) ) {
+      $this->_username = self::USERNAME_DEFAULT;
+    }
+    if ( empty( $this->_password ) ) {
+      $this->_password = 'ANY-PASSWORD';
```

Comment:
Is it ok to pass a non-null password values. Will this work when authentication is really enabled?

Reply:
I'm unsure, there's very little documentation on any of this... I'm basically piecing together what needs to be done with Thrift calls using just whatever is exposed in the interfaces and looking through HiveServer2/Impala server code to figure out what can be done. In theory this change only sets a random password if none is already passed in which fixes an invalid use case of setting a username but no password with SASL in Hive connections. I'm not sure what happens when secured auth is actually turned on. We'll need to test when that happens I guess.